src/statistic/statistic.js

```diff
@@ -0,0 +1,131 @@
+     */
+    discussionFormat(aDiscussion){
+        let time;
+        // If the finishTime is define we will calculate the duration of the deabte
+        if(typeof aDiscussion.finishTime !== 'undefined'){
+            let differenceHours = aDiscussion.finishTime.getHours() - aDiscussion.startTime.getHours();
+            let differenceMiutes = aDiscussion.finishTime.getMinutes() - aDiscussion.startTime.getMinutes();
+            time = differenceHours + "h " + differenceMiutes + "m";
+        }
+        else{
+            time = "en cours";
+        }
+        return {
+            id: aDiscussion._id,
+            title: aDiscussion.title,
+            description: aDiscussion.description,
+            time: time
+        }
+    }
+
+    /**
+     * Will format the question to have only useful data for statistics
+     * @param aQuestion an object Question that is a Question contained in the database
+     * @returns {{title: *, numberVotes: *}} array containing title: the title of the question and numberVotes: the total
+     * of all the votes for the responses for this question
+     */
+    questionFormat(aQuestion){
+        logger.debug(`Formatting question : ${aQuestion}`);
+        return {
+            title: aQuestion.titreQuestion,
+            numberVotes: aQuestion.numberVotes
+        }
+    }
+
+    /**
+     * Will format the response to have only useful data for statistics
+     * @param aResponse an object Response that is a Response contained in the database
+     * @param numberTotalVotes the number of votes for the response
+     * @returns {{response: *, percentage: number, numberVotes: *}}
+     */
+    responseFormat(aResponse, numberTotalVotes){
+        return {
+            response: aResponse.response,
+            numberVotes: aResponse.devices.length,
+            percentage: Math.round((aResponse.devices.length/numberTotalVotes) * 100)
+        }
+    }
+
+    /**
+     * Return all the informations desired when we want to make stats for an admin
+     * @param aUsername String that represents the username of the admin
+     * @returns {Promise<*[]>} an Array containing all the stats of the admin
+     */
+    async adminStats(aUsername){
+        let allDiscussions = await dbManager.getDiscussionsAdmin(aUsername);
+
+        return [allDiscussions.length, Array.from(allDiscussions.values(), d => this.discussionFormat(d))];
+    };
+
+    /**
+     * Get the number of the total of votes for all the responses for a question.
+     * @param aQuestionId integer that is the id of the question
+     * @param aDiscussionId integer that is the id of the discussion
+     * @returns {Promise<number>} an integer that is the number of votes
+     */
+    async getNumberVotesQuestion(aQuestionId, aDiscussionId) {
+        // Get all the responses related to the question desired
+        let allResponses = await dbManager.getResponsesQuestion(aQuestionId, aDiscussionId);
```

Comment:
Can `dbManager.getResponsesQuestion` return a null object ? If that's the case we need to verify the value of `allResponses` before using it.

Reply:
Yes it was effectively a case that I didn't implement I changed it in the latest commit